quill-jdbc/src/main/scala/io/getquill/sources/jdbc/TransactionalJdbcSource.scala

```diff
@@ -0,0 +1,31 @@
+package io.getquill.sources.jdbc
+
+import java.sql.Connection
+
+import io.getquill.JdbcSourceConfig
+import io.getquill.naming.NamingStrategy
+import io.getquill.sources.sql.idiom.SqlIdiom
+
+import scala.util.control.NonFatal
+
+class TransactionalJdbcSource[D <: SqlIdiom, N <: NamingStrategy, T](config: JdbcSourceConfig[D, N], conn: Connection)
+  extends JdbcSource[D, N](config) {
+
+  conn.setAutoCommit(false)
+
+  protected def connection = conn
+
+  def execute(f: JdbcSource[D, N] => T) = {
```

Comment:
I'd prefer to move this to `JdbcSource`, the indirection calling `execute` that uses `this` seems odd to me.